src/core/dfn.js

```diff
@@ -108,6 +114,13 @@ function computeType(dfn, linkingText) {
   // But other modules may end up adding a type (e.g., the WebIDL module)
 }
 
+function processAsTaskSource(dfn) {
+  // Task sources are exported by default.
+  if (!dfn.matches(".no-export, [data-noexport]")) {
+    dfn.dataset.export = "";
+  }
+}
+
 // Deal with export/no export
 function computeExport(dfn) {
   switch (true) {
```

Comment:
I'm not sure how I feel about this pattern 🤔😂

Reply:
At first, I was like... that's weird/not right... but now I really love how it split up the code into different cases. 

Eventually, once ECMAScript gets pattern matching, we can switch them all over to that:
https://github.com/tc39/proposal-pattern-matching